### DISCUSSION TITLE
Adding RTL_433 suppport to gateway, architecture discussion

### DISCUSSION THREAD

**Discussion by 1technophile** [theengs/gateway#216] [Ideas]
This would enable having one addon in HA for BLE and RTL_433 (instead of 3 currently), and would also allow RTL_433 easier support in Jeedom
and OpenHAB.

This topic is to focus on the architecture discussion of this, taking into account that RTL_433 is developed in C.

**Comment by 1technophile** [comment]
I'm thinking that the same docker could host Theengs Gateway (python) and RTL_433 , this would mean merging the docker files into one that does the 2 integrations.
https://github.com/pbkhrv/rtl_433-hass-addons/blob/main/rtl_433/Dockerfile
https://github.com/mihsu81/addon-theengsgw/blob/main/theengsgateway/Dockerfile

**Reply by catduckgnaf** [reply]
I am happy to help from the rtl_433 side. I can tell you currently that rtl_433 container is not being maintained, and the person working on it doesn't have any rf devices. I have made a new container add-on with a new and better template: 

https://github.com/catduckgnaf/rtl_433_haos_addon

I am happy to help. I think we could start by leaving the images separate, and configuring rtl_433 to send things to omg.

The issue is also very much discovery, and adding new devices, and that is the part where I believe OMG can help greatly.

**Reply by catduckgnaf** [reply]
I started laying the ground work, https://github.com/catduckgnaf/all-the-theengs

**Reply by DigiH** [reply]
> The issue is also very much discovery, and adding new devices, and that is the part where I believe OMG can help greatly.

Unfortunately with no distinction of rtl_433 key naming convention depending on their value types, good and complete discovery for rtl_433 devices is very hard, if not impossible, to achieve without meticulously investigating each individual decoder :( and again with every new or changed decoder.

**Reply by DigiH** [reply]
https://github.com/merbanan/rtl_433/discussions/3103

**Comment by catduckgnaf** [comment]
I think we can start by making things simple, We can install rtl_433 as a docker image and then simply run a pip command in the container setup to install rtl_433 for the moment. They can then talk to eachother. I know there is also the OMG on the ESP32 RTL project. 

Can you maybe take a look at the config file, and see the best way to get data into Theengs? (i also like the name)
